From 97cb83179f200937642d749f3ce7be9421437bd5 Mon Sep 17 00:00:00 2001
From: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Date: Wed, 13 Apr 2016 16:54:02 -0400
Subject: [PATCH 1036/1110] drm/amd/dal: Obtain stream encoders number in
 proper way.

Stream encoders number was hard coded to 6 while Baffin has 5.
This would lead to using the 6th invalid encoder for daisy chained
second display.

Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c b/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c
index c1fb291..5cf61bd 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce112/dce112_resource.c
@@ -1222,7 +1222,7 @@ bool dce112_construct_resource_pool(
 
 	pool->pipe_count =
 		dal_adapter_service_get_func_controllers_num(adapter_serv);
-	pool->stream_enc_count = 6;
+	pool->stream_enc_count = dal_adapter_service_get_stream_engines_num(adapter_serv);
 	pool->scaler_filter = dal_scaler_filter_create(ctx);
 	if (pool->scaler_filter == NULL) {
 		BREAK_TO_DEBUGGER();
-- 
2.7.4

